src/HIE/Bios/Environment.hs

```diff
@@ -167,6 +167,20 @@ addCmdOpts cmdOpts df1 = do
        liftIO $ hPutStrLn stderr "cannot set package flags with :seti; use :set"
     -}
 
+makePathsAbsolute :: FilePath -> DynFlags -> DynFlags
+makePathsAbsolute work_dir df = 
+  mapOverIncludePaths (prependIfNotAbsolute work_dir)
+  $ df
+    { importPaths = map (prependIfNotAbsolute work_dir) (importPaths df)
+    , libraryPaths  = map (prependIfNotAbsolute work_dir) (libraryPaths df)
+    , hpcDir = prependIfNotAbsolute work_dir (hpcDir df)
+    , frameworkPaths  = map (prependIfNotAbsolute work_dir) (frameworkPaths  df)
```

Comment:
Maybe this overly eager? I am actually not sure if this would have unexpected problems.